File: fizzbuzz/fizzbuzz.lua
local total = (arg[1] or 100)
local divs = { { 15, "fizzbuzz" }, { 3, "fizz" }, { 5, "buzz" } }

for n = 1, total do
  for i, _ in ipairs(divs) do
    if n % divs[i][1] == 0 then
      print(divs[i][2])
      goto skip
    end
  end

  print(n)

  ::skip::
end

File: fizzbuzz/fizzbuzz.ts
const Input: HTMLElement | null = document.getElementById("input");
const Output: HTMLElement | null = document.getElementById("output");
const Run: HTMLElement | null = document.getElementById("run");
const divs: ReadonlyArray<[number, string]> = [[ 15, "fizzbuzz" ], [3, "fizz"], [5, "buzz"]];

function fizzbuzz(total: number) {
    let result: Array<string> = [];

    back: for (let i = 1; i <= total; i++) {
        for (let j = 0; j < divs.length; j++) {
            if (i % divs[j][0] == 0) {
                result.push(divs[j][1]);
                continue back;
            }
        }

        result.push(i.toString());
    }

    if (Output != null) {
        Output.innerHTML = result.join('\n');
        Output.setAttribute("rows", total.toString());
    } else {
        console.log(result.join('\n'));
    }
}
